### DISCUSSION TITLE
face 0 area does not match neighbour by 37.19269211716968% -- possible face ordering problem. [Rotor 37]

### DISCUSSION THREAD

**Discussion by Mohsin-Bashir** [mdolab/dafoam#671] [General]
https://github.com/mdolab/dafoam/discussions/576#discussioncomment-8227188, as suggested in this link, I have replaced this line xs = self.DVGeo.update(self.ptSetName, config=None) by my own function, which provides deformed surface point coordinates. I have tested the whole thing earlier with Onera M6 test case. And it worked perfectly fine. Now I'm doing the same thing for Rotor 37 but getting an error, (more details in this file ([logDeformation.txt](https://github.com/user-attachments/files/16669728/logDeformation.txt)).

I have checked, the xs deformed surface points are passed correctly in correct order for the design surface which is in this case hub, shroud and blade.

What could be the problem? Any assistance would be greatly appreciated in identifying the issue.

**Update:**
Ok I just ran the rotor37 tutorial case, and getting the same error. Here is the [logOpt.txt](https://github.com/user-attachments/files/16670818/logOpt.txt). 
Its probably because I have more than one design surfaces. But If there is only "blade" as design surface then it cannot calculate sensitivity map what should I do?

**Comment by friedenhe** [comment]
It is hard to debug without more information, especially your case setup. However, it is likely to be caused by having multiple wall surfaces (in the M6 case, you had only one wall surface: wing, but in R37, you have multiple). So, a temporary workaround is to merge all wall surfaces as one.

**Reply by Mohsin-Bashir** [reply]
I specifically need the surface sensitivity map for the rotor 37 blade, excluding the hub and shroud. If I merge all the wall surfaces into a single entity, it will prevent me from identifying which surface sensitivities correspond to which specific values. Is there any solution to this issue?

**Reply by Mohsin-Bashir** [reply]
Would it be possible to modify the code so that the design surfaces and the all-wall surfaces don't have to be the same? This way, I can simply add one design surface, namely the blade, and run the optimization with a sensitivity map focused solely on the blade.

**Reply by Mohsin-Bashir** [reply]
Update: I merged all the wall surfaces and ran the rotor 37 tutorial case. However, by the second iteration, I encountered the same error as mentioned in the title of this discussion, though the occurrence was lower (around 1%). I then increased the matchTolerance from 0.0001 to 0.2, but the issue persisted, with the percentage increasing with each iteration. By the third iteration, it had risen to over 10%.

**Comment by friedenhe** [comment]
It seems that the problem may not be caused by multiple surfaces. Can you write the volume mesh (after applying your DIY surface mesh deformation and volume mesh warping) to the disk and load the mesh into Paraview to check why the periodic patches do not match?
